server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -121,6 +122,10 @@ public CreateDataStreamClusterStateUpdateRequest(String name,
     static ClusterState createDataStream(MetadataCreateIndexService metadataCreateIndexService,
                                          ClusterState currentState,
                                          CreateDataStreamClusterStateUpdateRequest request) throws Exception {
+        if (currentState.nodes().getMinNodeVersion().compareTo(Version.V_8_0_0) < 0) {
```

Comment:
I think check is a bit clearer: `currentState.nodes().getMinNodeVersion().before(Version.V_8_0_0)`